File: unibo.actor22/src/unibo/actor22comm/ActorForReply.java
package unibo.actor22comm;
import it.unibo.kactor.IApplMessage;
import unibo.actor22.Qak22Context;
import unibo.actor22.QakActor22;
import unibo.comm22.interfaces.IApplMsgHandler;
import unibo.comm22.interfaces.Interaction2021;
import unibo.comm22.utils.ColorsOut;
 

public class ActorForReply extends QakActor22{
private IApplMsgHandler h;
private Interaction2021 conn;

	public ActorForReply(String name, IApplMsgHandler h, Interaction2021 conn) {
		super(name);
		this.h = h;
		this.conn = conn;		 
	}

	@Override
	protected void handleMsg(IApplMessage msg) {
		//BasicUtils.aboutThreads(getName()  + " |  Before doJob - ");
		//ColorsOut.outappl( getName()  + " | handleMsg " + msg, ColorsOut.BLUE);
		if( msg.isReply() ) h.sendAnswerToClient(msg.toString(), conn);		
		Qak22Context.removeActor(this);
	}

}

File: unibo.qakactor22/src/main/kotlin/ActorForReply.kt
package it.unibo.kactor

import unibo.comm22.interfaces.IApplMsgHandler
import unibo.comm22.interfaces.Interaction2021
import unibo.comm22.utils.ColorsOut

class ActorForReply( name:  String,
                     val h: IApplMsgHandler, val conn: Interaction2021 ) : ActorBasic( name ) {

    override suspend fun actorBody(msg: IApplMessage) {
        ColorsOut.outappl(name + " | actorBody $msg conn= $conn", ColorsOut.MAGENTA);
        if( msg.isReply() ) h.sendAnswerToClient(msg.toString(), conn);
        context!!.removeInternalActor(this);
    }

}